CONTRIBUTING.md

```diff
@@ -200,6 +201,47 @@ Note that `git rebase` rewrites the commit history. You should avoid `git
 rebase` after you asked someone to review your code because the reviewer needs
 additional steps to ensure the review result is included.
 
+### Code style
+
+We use a style for source files based on [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html)
+except some cases, notably brace wrapping. Here is a brief list of the styles.
+
+- Use `snake_case`, not `CamelCase`
+- Use spaces. The indent width is four
+- Use `\r\n`, or `CR + LF` for line breaks
+- Use `\\` for inline comments. Use `/* */` for multi line comments after an
+  empty line
+- Break before braces in *most cases* (functions, conditionals, control
+  statements, etc)
+- Always check given arguments
+- Always check return code, return value, or `errno`
+- Return `esp_err_t` from functions where possible
+- Document public functions, data types, and macros in header files
+
```

Comment:
- For type names use the suffix `_t`: `typedef int my_type_t`, for function types `_cb_t`: `void (*my_callback_cb_t)()`, for structs `_s`: `struct my_struct_s`. Type names must be in snake-case too.
- Macros must be in `UPPER_CASE`

Reply:
fixed